### PR TITLE
Adds new new affordance: placeable and new atomic task: place upright

### PR DESCRIPTION
**Pull request by rebeccazhang0707**
## Summary
Adds new new affordance: **placeable** and new atomic task: **place upright task**

## Detailed description
- What was the reason for the change?
    - Place upright task is a common inhouse task for humanoids. We believe '**placeable**' should be a basic affordance to support. So we add related atomic task and example to showcase how to use this affordance.
- What has been changed?
    - adds new atomic task: place upright task (and a test script for this task)
    - adss new affordance: placeable (placeable object can be placed upright, like mug, bottle, etc.)
    - adds new example: agibot left_arm place upright mug
    - adds a unitest for 'place_upright_task'
- What is the impact of this change?
    - fixed the bug in agibot.py: since SceneCfg and ActionsCfg cannot accept a new type of property like ArmMode in manager-based workflow

## Test Pipeline
1. Zero Action Policy:
`python isaaclab_arena/examples/policy_runner.py --policy_type zero_action tabletop_place_upright --object mug`

2. Record demos:
`python isaaclab_arena/scripts/imitation_learning/record_demos.py --dataset_file datasets/dataset_agibot_left_arm_rel.hdf5 --num_demos 1 tabletop_place_upright --teleop_device keyboard`

3. Replay demos:
`python isaaclab_arena/scripts/imitation_learning/replay_demos.py --dataset_file datasets/dataset_agibot_left_arm_rel.hdf5 tabletop_place_upright`

4. Test place_upright_mug task:
`pytest isaaclab_arena/tests/test_place_upright_mug.py`

### CODE REVIEW

File: isaaclab_arena/affordances/placeable.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        world_up[..., 2] = 1.0
+
+        cos_angle = torch.sum(upright_axis_world * world_up, dim=-1).clamp(-1.0, 1.0)
+        angle_error = torch.acos(cos_angle)
+
+        orientation_threshold_tensor = torch.as_tensor(
+            orientation_threshold, device=object_quat.device, dtype=object_quat.dtype
+        )
+
+        success = angle_error < orientation_threshold_tensor
```
Comment (viiik-inside) [inline isaaclab_arena/affordances/placeable.py:72]: nice

File: isaaclab_arena/affordances/placeable.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+class Placeable(AffordanceBase):
+    """Interface for placeable objects.
+    Placeable objects are objects that can be placed upright in a scene.
+    They are characterized by their upright axis and orientation threshold, which are used to determine if the object is placed upright.
+    - The upright axis is the axis in the object's local frame that is used to determine if the object is placed upright.
+    - The orientation threshold is the threshold for the angle between the upright axis and the world +Z direction.
+    """
+
+    def __init__(self, upright_axis_name: Literal["x", "y", "z"] = "z", orientation_threshold: float = 0.5, **kwargs):
```
Comment (alexmillane) [inline isaaclab_arena/affordances/placeable.py]: suggestion to add a unit to `orientation_threshold`. For example `orientation_threshold_deg` (if indeed the unit here is degrees.

File: isaaclab_arena/affordances/placeable.py (hunk 1 above)
Comment (alexmillane) [reply isaaclab_arena/affordances/placeable.py:72]: yea this is cool!

File: isaaclab_arena/affordances/placeable.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        env: ManagerBasedEnv,
+        env_ids: torch.Tensor | None,
+        asset_cfg: SceneEntityCfg | None = None,
+        upright_percentage: float = 1.0,
+    ):
+        """Place the object upright (in all the environments).
+        This function places the object upright by rotating its upright axis towards the world +Z direction.
+        The upright percentage is a value in [0, 1] that describes the absolute target angle between the upright axis and the world +Z direction:
+        0.0 results in a perpendicular orientation (90 degrees) and 1.0 fully aligns the axis with +Z (0 degrees).
+        """
```
Comment (alexmillane) [inline isaaclab_arena/affordances/placeable.py:100]: In the case of `upright_percentage = 0.0` which world axis does the object's upright axis point along?

File: isaaclab_arena/affordances/placeable.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    object_entity.write_root_pose_to_sim(pose_tensor, env_ids=env_ids)
+    zero_velocity = torch.zeros((env_ids.numel(), 6), device=device, dtype=dtype)
+    object_entity.write_root_velocity_to_sim(zero_velocity, env_ids=env_ids)
+
+
+def _compute_target_quaternions(
+    object_quat: torch.Tensor,
+    upright_percentage: float | torch.Tensor,
+    upright_axis_name: str,
+) -> torch.Tensor:
```
Comment (alexmillane) [inline isaaclab_arena/affordances/placeable.py:208]: Suggestion to add a docstring.

What's the expected shape of `upright_percentage` if it's a tensor? I would guess: `[num_envs, 1]`?

File: isaaclab_arena/affordances/placeable.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    """Compute the target quaternions for the object given the current orientation and the upright percentage."""
+
+    upright_percentage_t = torch.as_tensor(upright_percentage, device=object_quat.device, dtype=object_quat.dtype)
+    if upright_percentage_t.dim() == 0:
+        upright_percentage_t = upright_percentage_t.expand(object_quat.shape[0])
+
+    axis_index = {"x": 0, "y": 1, "z": 2}[upright_axis_name]
+    rotation_mats = math_utils.matrix_from_quat(object_quat)
+    current_axis = rotation_mats[:, :, axis_index]
```
Comment (alexmillane) [inline isaaclab_arena/affordances/placeable.py]: This code is repeated from above. Suggestion to move to a function

```
def get_object_axis_in_world_frame(object_quat: torch.tensor) -> torch.tensor:
    axis_index = {"x": 0, "y": 1, "z": 2}[upright_axis_name]
    rotation_mats = math_utils.matrix_from_quat(object_quat)
    current_axis = rotation_mats[:, :, axis_index]
    return current_axis
```

File: isaaclab_arena/tests/test_place_upright_task.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        _test_place_upright_mug_condition,
+        headless=HEADLESS,
+    )
+    assert result, f"Test {_test_place_upright_mug_condition.__name__} failed"
+
+
+if __name__ == "__main__":
+    # test_place_upright_mug_single()
+    test_place_upright_mug_multi()
+    # test_place_upright_mug_condition()
```
Comment (alexmillane) [inline isaaclab_arena/tests/test_place_upright_task.py]: Thanks for adding these tests!

Suggestion to uncommen the lines here.

File: isaaclab_arena_environments/tabletop_place_upright_environment.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+                    "pose_range": {
+                        "x": (-0.05, 0.2),
+                        "y": (-0.10, 0.10),
+                        "z": (0.75, 0.75),
+                        "roll": (-1.57, -1.57),
+                        "yaw": (-0.57, 0.57),
+                    },
+                    "asset_cfgs": [SceneEntityCfg("mug")],
+                },
+            )
```
Comment (alexmillane) [inline isaaclab_arena_environments/tabletop_place_upright_environment.py:50]: @viiik-inside I feel like we need to add an option to our objects to add randomness to the poses on reset. I've now seen this pattern in a few MRs where people are overriding the `EventCfg` coming from the task. This kind of breaks our design. Maybe we could expand our definition of setting an objects pose to take either a `Pose` or a `PoseRange` such that `object.set_initial_pose(Pose | PoseRange)`. In the case of a pose range, it would an event to the `Cfg`.

**Review by alexmillane — APPROVED** [review]
Thanks for adding this. Look good to me!

I have a few comments. Feel free to merge once you have addressed them.

File: isaaclab_arena/affordances/placeable.py (hunk 2 above)
Comment (rebeccazhang0707) [reply isaaclab_arena/affordances/placeable.py]: thanks Alex, here is radians actually, and I will revise the default value to  `math.pi / 18.0` to indict that this is in radians, and adds the unit is radians in the comments of this method.

File: isaaclab_arena/affordances/placeable.py (hunk 3 above)
Comment (rebeccazhang0707) [reply isaaclab_arena/affordances/placeable.py:100]: upright_percentage = 0.0: the target upright axis is perpendicular to world +Z (90°), and the direction with the smallest angle to the object_upright_axis, in the plane spanned by the object_upright_axis and world +Z.
